Makefile

```diff
@@ -1,9 +1,40 @@
+PROJECT := liftbridge-api
+
+BUF_VERSION := 0.31.1
+
+UNAME_OS := $(shell uname -s)
+UNAME_ARCH := $(shell uname -m)
+CACHE_BASE := $(HOME)/.cache/$(PROJECT)
+CACHE := $(CACHE_BASE)/$(UNAME_OS)/$(UNAME_ARCH)
+CACHE_BIN := $(CACHE)/bin
+CACHE_VERSIONS := $(CACHE)/versions
+
+export PATH := $(abspath $(CACHE_BIN)):$(PATH)
+
+BUF := $(CACHE_VERSIONS)/buf/$(BUF_VERSION)
+$(BUF):
+	@rm -f $(CACHE_BIN)/buf
+	@mkdir -p $(CACHE_BIN)
+	curl -sSL \
+		"https://github.com/bufbuild/buf/releases/download/v$(BUF_VERSION)/buf-$(UNAME_OS)-$(UNAME_ARCH)" \
+		-o "$(CACHE_BIN)/buf"
+	chmod +x "$(CACHE_BIN)/buf"
+	@rm -rf $(dir $(BUF))
+	@mkdir -p $(dir $(BUF))
+	@touch $(BUF)
+
 .PHONY: all
 all: go py
 
 .PHONY: go
```

Comment:
Just a cleanup: I'd recommend doing a `rm -rf go` before the `protoc` call, in case you ever rename `api.proto`, or split it into multiple files (which I'd also recommend).